### DISCUSSION TITLE
Need help as beginer how to add tempreture sensor lm35 and set and get data in edgex foundry.

### DISCUSSION THREAD

**Discussion by humendra1007** [edgexfoundry/community#300] [Q&A]
if you guys have any tutorial pdf or video or link please share.

**Comment by jiekechoo** [comment]
No, there is no docs for sensor LM35.
You should use a smart device embedded the sensor, such as a serial interface or modbus.

**Comment by chc69** [comment]
@humendra1007 when you said LM35, do you mean the TI IC [Link](https://www.ti.com/lit/ds/symlink/lm35.pdf?ts=1723170602779&ref_url=https%253A%252F%252Fwww.ti.com%252Fproduct%252Fzh-tw%252FLM35%253Fbm-verify%253DAAQAAAAJ_____0eF6nl1-ynNShyzILuhq27jy_r0FpYCBWunrnezdjbHiE0blGbfvzHuimh-1NYVfuuBnZKHK7oG3FxpH6sbr1fdMqUfTA52I6F7V1zJJCLr10LtBq_6QlLgSwjUoULgltcmArDgoHt_YuCLRpq_fU_27PRo06GkD6yDYoqxojSUxEi7K0THuH3XZj8PHCYx0HWz1kUpnvDYcU5LFxB8hbmkwnSNcLy6742uvG-j10Dh-ZDsXNlvuBDSCCMuL-eIsKCjfDh2FJ2SaPxN8JouCSVOA4jyynmBD6O1BRykmawuCsE7NCZo)

EdgeX's role is an IoT gateway and edge computing platform, EdgeX retrieves data by several industrial protocols e.g. Modbus, so it won't be able to read data from a device/component without protocol. 
You can find such a digital temperature sensor from internet, e.g. 
[Link](https://www.electronicpro.co.za/products/rs485-modbus-rtu-temperature-and-humidity-transmitter-temperature-sensor).
